### PR TITLE
fix: fix the problem that can't accept the notify of block on-chain

### PR DESCRIPTION
**Pull request by sulenn**
fix the problem that can't accept the notify of block on-chain after reconnet channel network

### CODE REVIEW

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- conn/channel.go  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=6036981&bid=19782069)

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- conn/channel.go  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=6036981&bid=19782069)

File: conn/channel.go [hunk 1]
```diff
@@ -962,11 +962,16 @@ func (hc *channelSession) processMessages() {
 				}
 				hc.c = con
 				hc.closed = make(chan interface{})
+				hc.nodeInfo.Protocol = 1
```
Comment (bxq2011hust) [inline conn/channel.go:965]:
```suggestion
				
```

File: conn/channel.go [hunk 2]
```diff
@@ -962,11 +962,16 @@ func (hc *channelSession) processMessages() {
...
+				go hc.processMessages()
```
Comment (bxq2011hust) [inline conn/channel.go:966]:
```suggestion

```

File: conn/channel.go [hunk 3]
```diff
@@ -962,11 +962,16 @@ func (hc *channelSession) processMessages() {
...
+				if err = hc.handshakeChannel(); err != nil {
+					fmt.Printf("re-handshake channel protocol failed, use default protocol version")
+				}
 				err = hc.sendSubscribedTopics() // re-subscribe topic
 				if err != nil {
 					log.Printf("re-subscriber topic failed when successfully re-connect network")
 				}
-				break
+				return
```
Comment (bxq2011hust) [inline conn/channel.go:974]: break

File: conn/channel.go [hunk 4]
```diff
@@ -962,11 +962,16 @@ func (hc *channelSession) processMessages() {
 				}
 				hc.c = con
 				hc.closed = make(chan interface{})
+				hc.nodeInfo.Protocol = 1
+				go hc.processMessages()
+				if err = hc.handshakeChannel(); err != nil {
+					fmt.Printf("re-handshake channel protocol failed, use default protocol version")
```
Comment (bxq2011hust) [inline conn/channel.go:968]:
```suggestion
					fmt.Printf("handshake channel protocol failed, use default protocol version")
```

File: conn/channel.go [hunk 5]
```diff
@@ -962,11 +962,16 @@ func (hc *channelSession) processMessages() {
...
+				}
 				err = hc.sendSubscribedTopics() // re-subscribe topic
 				if err != nil {
 					log.Printf("re-subscriber topic failed when successfully re-connect network")
```
Comment (bxq2011hust) [inline conn/channel.go:972]:
```suggestion
					log.Printf("re-subscriber topic failed")
```

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- conn/channel.go  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=6036981&bid=19782069)

**Review by bxq2011hust — APPROVED** [review]

**Pull request merged**